### ISSUE TITLE
Vulnerabilities found after npm install

### ISSUE THREAD

**Issue by dangelion** [WeAreAthlon/frontend-webpack-boilerplate#3]
Hi
after running `npm install` I get this message:

```
added 1743 packages from 735 contributors and audited 15913 packages in 43.147s
found 13 vulnerabilities (1 low, 10 moderate, 2 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```

What's wrong?
Thanks

**Comment by pnikolov** [comment]
Hi,

Thanks for sharing your feedback. It turns out that there are _dev_ dependencies that have been reported to be vulnerable to _Arbitrary File Overwrite_ (`tar` - *reported 3 days ago https://github.com/sass/node-sass/issues/2625*)
All those are _dev_ dependencies of `node-sass` with _high_ severity but as those packages should not be part of your `production` code there is no actual risk.

Anyway I've reviewed the full `npm audit` report and have addressed all the reported issues and have just **released** [Release 2.4.1](https://github.com/WeAreAthlon/frontend-webpack-boilerplate/releases/tag/v2.4.1) - _Release Notes_

> * NPM audit review and dependency packages updates
>  * Auto fixing with `npm audit fix`
>  * Manually updated
>    * updated `tar` to `4.4.2` dependency path `node-sass > node-gyp > tar` - [more info](https://npmjs.com/advisories/803)
>    * updated `braces` to `2.3.2` dependency path `browser-sync > micromatch > braces` - [more info](https://npmjs.com/advisories/786)
> * _README_ file updated
>  * Added instructions how to update/migrate to newer version of this setup.

So to resolve all reported vulnerabilities update to **[`2.4.1`](https://github.com/WeAreAthlon/frontend-webpack-boilerplate/releases/tag/v2.4.1)** release and make a **clean install** after this by running:
```sh
$ npm run ci
```
More instructions you can find in the [README#clean-install](https://github.com/WeAreAthlon/frontend-webpack-boilerplate#clean-install-after-update-from-older-release-of-this-setup)

**Comment by pnikolov** [comment]
I've just republished released [Release 2.4.1](https://github.com/WeAreAthlon/frontend-webpack-boilerplate/releases/tag/v2.4.1) to correctly reference the updated `package.json` version. So feel free to install / update.

**Comment by dangelion** [comment]
Hi @pnikolov 
thanks for the fast answer! It solved. 
Hope you're going to maintains this boilerplate for a long time, we're considering to adopt it for our projects. 👍

**Comment by dangelion** [comment]
Hi @pnikolov 
I downloaded new release 2.4.2 and after `npm install` I get again:
```
found 2 vulnerabilities (1 low, 1 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```
Let me know
Thanks

**Comment by pnikolov** [comment]
Thanks for the feedback. I was not aware that after updating the package version it will reintroduce the  official latest versions of the referenced dependencies trees. I've release [v2.4.3](https://github.com/WeAreAthlon/frontend-webpack-boilerplate/releases/tag/v2.4.3) and have additionally simulated clean install and here is the result:
```sh
added 1774 packages from 738 contributors and audited 15992 packages in 50.103s
found 0 vulnerabilities
```
**Note:** We are still waiting for a new release of [`node-sass`](https://github.com/sass/node-sass) that will not rely on the outdated `tar` and will simplify our release process.

**Comment by pnikolov** [comment]
I've just released [Release `2.5.0`](https://github.com/WeAreAthlon/frontend-webpack-boilerplate/releases/tag/v2.5.0)

The main reason of this is the behaviour of `npm install` command. Every time you execute `npm install` it will rewrite your `package-lock.json` file and will reintroduce the current security issues inherited from the development packages of the package dependency (`tar` from `node-sass > node-gyp > tar` and `braces` from `browser-sync > micromatch > braces`).

So, from now on use only the `npm ci` command for setup/installation. It will respect the `package-lock.json` file and will not overwrite it. More on the clean install npm command can be read yere [`npm ci`](https://docs.npmjs.com/cli/ci.html)

**Comment by kyds3k** [comment]
After running `npm ci`, I get:

`28 vulnerabilities (2 low, 3 moderate, 23 high)`

Should I ignore this or what is the recommended action?
